tests/h/services/search_index/_queue_test.py

```diff
@@ -137,206 +136,192 @@ def add_where(self, queue):
 
 class TestSync:
     def test_it_does_nothing_if_the_queue_is_empty(self, batch_indexer, queue):
-        counts = queue.sync(LIMIT)
+        counts = queue.sync(1)
 
         assert counts == {}
         batch_indexer.index.assert_not_called()
 
     def test_it_ignores_jobs_that_arent_scheduled_yet(
-        self, add_all, batch_indexer, queue
+        self, batch_indexer, now, queue, SyncAnnotationJobFactory
     ):
-        add_all(schedule_in=ONE_WEEK_IN_SECONDS)
+        SyncAnnotationJobFactory(scheduled_at=now + datetime_.timedelta(hours=1))
 
-        counts = queue.sync(LIMIT)
+        counts = queue.sync(1)
 
         assert counts == {}
         batch_indexer.index.assert_not_called()
 
-    @pytest.mark.usefixtures("with_queued_annotations")
     def test_it_ignores_jobs_beyond_limit(
-        self, all_annotation_ids, batch_indexer, queue
+        self, batch_indexer, queue, SyncAnnotationJobFactory
     ):
-        queue.sync(LIMIT)
+        limit = 1
+        SyncAnnotationJobFactory.create_batch(size=limit + 1)
```

Comment:
The tests are now more repetitive: each test makes its own call(s) to `SyncAnnotationJobFactory()`. But I think that's highly preferable to having all the tests quite invisibly depend on a complex of shared fixtures like they did before. It's better if each test does its own test set up in the test itself, that makes the test easier to understand. And the flexible `SyncAnnotationJobFactory` fixture just makes that set up really easy to write.

(I'm not suggesting this as a universal rule, in many cases it is useful for tests to depend on fixtures for set up, often with a `@pytest.mark.usefixtures`.)